include/bout/sys/expressionparser.hxx

```diff
@@ -127,7 +127,7 @@ private:
     signed char curtok;  ///< Current token. -1 for number, -2 for string, 0 for "end of input"
     double curval; ///< Value if a number
     std::string curident; ///< Identifier, variable or function name
-    signed char LastChar;   ///< The last character read from the string
+    char LastChar;   ///< The last character read from the string
```

Comment:
This was originally part of me trying to fix silent conversions with minimal changes, but I ended up having to explicitly cast every call to `stringstream::get()` anyway, so this change can be undone.